wonnx/src/compiler.rs

```diff
@@ -460,28 +459,65 @@ pub fn compile(
                 }
             }
 
-            if axis >= (input_shapes[0].rank() as i64) {
+            if axis >= (input_shapes[0].rank() as i64) || axis < 0 {
                 return Err(CompileError::InvalidAttributeValue {
                     attribute: "axis".to_string(),
                     value: format!("{}", axis),
                     opset_version,
                 });
             }
 
-            if axis != 1 {
-                return Err(CompileError::UnimplementedVariant {
-                    variant: format!(
-                        "softmax on an axis ({}) other than the second with [1,n] inputs",
-                        axis,
-                    ),
-                    op: "Softmax".to_string(),
-                });
-            }
+            let left_of_axis = input_shapes[0].dims[0..(axis as usize)]
+                .iter()
+                .product::<u64>();
+            let axis_chunk = input_shapes[0].dims[(axis as usize)..]
+                .iter()
+                .product::<u64>();
+            let right_of_axis_chunk = input_shapes[0].dims[((axis + 1) as usize)..]
```

Comment:
So in the case that the axis is the last axis. Wouldn't that overflow?

Reply:
Ok. So I just checked and it won't because if axis +1 = rank(). this result in a empty list which product is 1.